specs/lib/gltfPipelineSpec.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var clone = require('clone');
+var gltfPipeline = require('../../lib/gltfPipeline');
+var processJSON = gltfPipeline.processJSON;
+var processJSONWithExtras = gltfPipeline.processJSONWithExtras;
+var processJSONToDisk = gltfPipeline.processJSONToDisk;
+var processFile = gltfPipeline.processFile;
+var processFileToDisk = gltfPipeline.processFileToDisk;
+var readGltf = require('../../lib/readGltf');
+var removePipelineExtras = require('../../lib/removePipelineExtras');
+var addPipelineExtras = require('../../lib/addPipelineExtras');
+var writeBinaryGltf = require('../../lib/writeBinaryGltf');
+
+var gltfPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.gltf';
+var glbPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.glb';
+var outputPath = './output/';
+
+describe('gltfPipeline', function() {
+    it('optimizes a gltf JSON given an embedded JSON', function() {
+        var gltfCopy;
+        var options = {};
+        readGltf(gltfPath, function(gltf) {
+            gltfCopy = clone(gltf);
+            processJSON(gltf, options, function(gltf) {
+                expect(gltf).toBeDefined();
+                expect(gltf).not.toBe(gltfCopy);
+            });
+        });
+    });
+    
+    it('optimizes a gltf JSON given a resourcePath', function() {
+        var gltf;
+        var gltfCopy;
+        var options = { 'resourcePath' : path.dirname(gltfPath) };
+        fs.readFile(gltfPath, function(err, data) {
```

Comment:
readGltf includes the `loadGltfUris' step, and this test is checking that `processJSON` properly handling a JSON that has yet to have that step applied